internal/services/privatednsresolver/private_dns_resolver_forwarding_rule_resource.go

```diff
@@ -0,0 +1,335 @@
+package privatednsresolver
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/dnsresolver/2022-07-01/dnsforwardingrulesets"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/dnsresolver/2022-07-01/forwardingrules"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type PrivateDNSResolverForwardingRuleModel struct {
+	Name                                     string                 `tfschema:"name"`
+	PrivateDNSResolverDnsForwardingRulesetId string                 `tfschema:"private_dns_resolver_dns_forwarding_ruleset_id"`
+	DomainName                               string                 `tfschema:"domain_name"`
+	Enabled                                  bool                   `tfschema:"enabled"`
+	Metadata                                 map[string]string      `tfschema:"metadata"`
+	TargetDnsServers                         []TargetDnsServerModel `tfschema:"target_dns_servers"`
+}
+
+type TargetDnsServerModel struct {
+	IPAddress string `tfschema:"ip_address"`
+	Port      int64  `tfschema:"port"`
+}
+
+type PrivateDNSResolverForwardingRuleResource struct{}
+
+var _ sdk.ResourceWithUpdate = PrivateDNSResolverForwardingRuleResource{}
+
+func (r PrivateDNSResolverForwardingRuleResource) ResourceType() string {
+	return "azurerm_private_dns_resolver_forwarding_rule"
+}
+
+func (r PrivateDNSResolverForwardingRuleResource) ModelObject() interface{} {
+	return &PrivateDNSResolverForwardingRuleModel{}
+}
+
+func (r PrivateDNSResolverForwardingRuleResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return forwardingrules.ValidateForwardingRuleID
+}
+
+func (r PrivateDNSResolverForwardingRuleResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"private_dns_resolver_dns_forwarding_ruleset_id": {
```

Comment:
seems like this should be
```suggestion
		"dns_forwarding_ruleset_id": {
```